core/src/main/kotlin/io/customer/sdk/core/di/SDKComponentExt.kt

```diff
@@ -9,8 +9,10 @@ import android.content.Context
 /**
  * Create and register an instance of AndroidSDKComponent with the provided context,
  * only if it is not already initialized.
+ * This function should be called from all entry points of the SDK to ensure that
+ * AndroidSDKComponent is initialized before accessing any of its dependencies.
  */
-fun SDKComponent.registerAndroidSDKComponent(
```

Comment:
Since the method is intended for internal SDK use only, I renamed it to provide more clarity on its purpose and function.